### PR TITLE
Reuse requests.session (for v4.1.11)

### PR DESCRIPTION
**Pull request by iAnanich**
sync PR #98 to v4.1.11

* rewrite `server_connector.ClientSession.request` method for easier monkey patching during testing
* fix type annotations

### CODE REVIEW

File: eyes_core/applitools/core/server_connector.py [hunk 1]
```diff
@@ -165,15 +229,21 @@ class ServerConnector(object):
 
     _is_session_started = False
 
-    def __init__(self):
-        # type: () -> None
+    def __init__(self, client_session=None):
+        # type: (ClientSession) -> None
```
Comment (Infernion) [inline eyes_core/applitools/core/server_connector.py]: need to change type annotation to Optional[ClientSession]

Comment (iAnanich) [reply eyes_core/applitools/core/server_connector.py]: got it! changed in the following commit

**Comment by iAnanich** [comment]
I saw new commits in develop, so changing branch to another (prepared one)

**Comment by iAnanich** [comment]
Merged prepared yesterday branch, so commit history looks a bit messy, but the idea is that it's in sync with v4.1.11

**Pull request merged**